src/prefect/core/task.py

```diff
@@ -659,6 +659,18 @@ def __call__(
             - Task: a new Task instance
         """
         new = self.copy(**(task_args or {}))
+
+        positional_args = [
+            name
+            for name, parameter in inspect.signature(new).parameters.items()
+            if parameter.kind == inspect.Parameter.POSITIONAL_ONLY
+        ]
+        if positional_args:
+            raise TypeError(
+                "Prefect passes arguments to task as keyword arguments. "
+                f"Positional-Only arguments found : {positional_args}"
+            )
```

Comment:
Could we check this at `__init__` time instead of `__call__` time so we raise the error at flow definition instead of runtime?

Reply:
Also, cc @tpdorsey I think we'll want to compose an error message here that is consistent with our style.

I'm imagining something like

> Found positional-only argument(s) in function signature for task {task_name}: {positional_args}. 
> Prefect passes arguments using keywords and does not support positional-only arguments.

Reply:
My suggested refinement of your suggestion, which I think works quite well:
 
"Found positional arguments in the function signature for task {task_name}: {positional_args}.
Prefect passes arguments using keywords and does not support positional-only arguments."

Reply:
Maybe following the PEP (https://peps.python.org/pep-0570/) we should say "positional-only" parameters. I don't think we can drop the "-only" from the first line.

Reply:
Agreed, follow PEP example: 

"Found positional-only arguments in the function signature for task {task_name}: {positional_args}.
Prefect passes arguments using keywords and does not support positional-only arguments."

Reply:
The document also refers to "parameters" instead of "arguments" so we may want to use that? I don't have a strong preference though.

Reply:
I do like to follow the PEP example when we can. However, looking at the Python docs, "argument" and "parameter" seem to be used interchangeably.  

If there is a distinction, I perceive it as: parameter is the definition of what you can pass, argument is what you pass as a value for that parameter. Maybe.

I don't have strong feelings about it either. I think the usage pattern is to use the term that promotes readability.

Reply:
I have used parameters throughout. It also prefer that parameters read appropriate for signature and arguments are more appropriate for the values supplied to the function. Below is a sample error message.

```python
python
Python 3.10.4 (main, Apr  8 2022, 17:35:13) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def foo(a, b, /): pass
... 
>>> foo(a=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() got some positional-only arguments passed as keyword arguments: 'a'
```

```
Traceback (most recent call last):
  File "/tmp/foo.py", line 4, in <module>
    def add(a, b, /):
  File "/home/karthikeyan/stuff/python/prefect/src/prefect/utilities/tasks.py", line 406, in <lambda>
    return lambda fn: prefect.tasks.core.function.FunctionTask(
  File "/home/karthikeyan/stuff/python/prefect/src/prefect/core/task.py", line 162, in init
    old_init(self, *args, **kwargs)
  File "/home/karthikeyan/stuff/python/prefect/src/prefect/tasks/core/function.py", line 50, in __init__
    super().__init__(name=name, **kwargs)
  File "/home/karthikeyan/stuff/python/prefect/src/prefect/core/task.py", line 162, in init
    old_init(self, *args, **kwargs)
  File "/home/karthikeyan/stuff/python/prefect/src/prefect/core/task.py", line 359, in __init__
    raise TypeError(
TypeError: Found positional-only parameters in the function signature for task add: ['a', 'b']. Prefect passes arguments using keywords and does not support positional-only parameters.
```